### PR TITLE
Feat :Add support for hard delete table entity.

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on Adding hard delete support for table entities and also added basic unit tests.

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/59080942/163197247-62053c9e-1144-4d1e-869b-c1c50020f3e8.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

**Review by ShaileshParmar11 — APPROVED** [review]
LGTM!

**Comment by harshach** [comment]
@Sachin-chaurasiya anti tests are failing
```022-04-13T16:35:43.0496186Z [INFO] PASS @openmetadata src/pages/teams/Form.test.tsx
2022-04-13T16:35:43.0508787Z [INFO]   ● Console
2022-04-13T16:35:43.0509106Z [INFO] 
2022-04-13T16:35:43.0509392Z [INFO]     console.error
2022-04-13T16:35:43.0509872Z [INFO]       Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
2022-04-13T16:35:43.0510333Z [INFO]       
2022-04-13T16:35:43.0510663Z [INFO]       Check the render method of `TeamsForm`.
2022-04-13T16:35:43.0511037Z [INFO]           in mockConstructor (created by TeamsForm)
2022-04-13T16:35:43.0511412Z [INFO]           in div (created by TeamsForm)
2022-04-13T16:35:43.0511755Z [INFO]           in div (created by TeamsForm)
2022-04-13T16:35:43.0512094Z [INFO]           in div (created by TeamsForm)
2022-04-13T16:35:43.0512429Z [INFO]           in div (created by TeamsForm)
2022-04-13T16:35:43.0512841Z [INFO]           in TeamsForm
2022-04-13T16:35:43.0513195Z [INFO]           in Router (created by MemoryRouter)
2022-04-13T16:35:43.0513807Z [INFO]           in MemoryRouter
2022-04-13T16:35:43.0514106Z [INFO] 
2022-04-13T16:35:43.0514635Z [INFO]       at printWarning (node_modules/react-dom/cjs/react-dom.development.js:88:30)
2022-04-13T16:35:43.0515218Z [INFO]       at error (node_modules/react-dom/cjs/react-dom.development.js:60:5)
2022-04-13T16:35:43.0515981Z [INFO]       at validateFunctionComponentInDev (node_modules/react-dom/cjs/react-dom.development.js:17578:9)
2022-04-13T16:35:43.0516670Z [INFO]       at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17545:7)
2022-04-13T16:35:43.0517299Z [INFO]       at beginWork (node_modules/react-dom/cjs/react-dom.development.js:18596:16)
2022-04-13T16:35:43.0517889Z [INFO]       at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23179:14)
2022-04-13T16:35:43.0518508Z [INFO]       at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
2022-04-13T16:35:43.0519113Z [INFO]       at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
2022-04-13T16:35:43.0519486Z [INFO] 
2022-04-13T16:35:43.7084669Z [INFO] PASS @openmetadata src/components/buttons/Button/Button.test.tsx
2022-04-13T16:35:44.4422678Z [INFO] PASS @openmetadata src/pages/tags/Form.test.tsx
2022-04-13T16:35:44.4432331Z [INFO]   ● Console
2022-04-13T16:35:44.4432655Z [INFO] 
2022-04-13T16:35:44.4432931Z [INFO]     console.error
2022-04-13T16:35:44.4433423Z [INFO]       Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
2022-04-13T16:35:44.4433858Z [INFO]       
2022-04-13T16:35:44.4434191Z [INFO]       Check the render method of `ForwardRef`.
2022-04-13T16:35:44.4434571Z [INFO]           in mockConstructor (created by ForwardRef)
2022-04-13T16:35:44.4434941Z [INFO]           in div (created by ForwardRef)
2022-04-13T16:35:44.4435283Z [INFO]           in div (created by ForwardRef)
2022-04-13T16:35:44.4435616Z [INFO]           in div (created by ForwardRef)
2022-04-13T16:35:44.4435966Z [INFO]           in div (created by ForwardRef)
2022-04-13T16:35:44.4436267Z [INFO]           in ForwardRef
2022-04-13T16:35:44.4436600Z [INFO]           in Router (created by MemoryRouter)
2022-04-13T16:35:44.4436931Z [INFO]           in MemoryRouter
2022-04-13T16:35:44.4449645Z [INFO] 
2022-04-13T16:35:44.4452618Z [INFO]       at printWarning (node_modules/react-dom/cjs/react-dom.development.js:88:30)
2022-04-13T16:35:44.4453235Z [INFO]       at error (node_modules/react-dom/cjs/react-dom.development.js:60:5)
2022-04-13T16:35:44.4453876Z [INFO]       at validateFunctionComponentInDev (node_modules/react-dom/cjs/react-dom.development.js:17578:9)
2022-04-13T16:35:44.4454568Z [INFO]       at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17545:7)
2022-04-13T16:35:44.4455756Z [INFO]       at beginWork (node_modules/react-dom/cjs/react-dom.development.js:18596:16)
2022-04-13T16:35:44.4456289Z [INFO]       at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23179:14)
2022-04-13T16:35:44.4456835Z [INFO]       at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
2022-04-13T16:35:44.4457375Z [INFO]       at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
2022-04-13T16:35:44.4457664Z [INFO] 
2022-04-13T16:35:45.0708911Z [INFO] PASS @openmetadata src/components/common/error-with-placeholder/ErrorPlaceHolder.test.tsx
2022-04-13T16:35:45.7550445Z [INFO] PASS @openmetadata src/components/common/error/Error.test.tsx
2022-04-13T16:35:46.3848866Z [INFO] PASS @openmetadata src/utils/SvgUtils.test.tsx
2022-04-13T16:35:47.0135800Z [INFO] PASS @openmetadata src/components/onboarding/Onboarding.test.tsx
2022-04-13T16:35:47.6408605Z [INFO] PASS @openmetadata src/components/Loader/Loader.test.tsx
2022-04-13T16:35:47.6428414Z [INFO] 
2022-04-13T16:35:47.6431025Z [INFO] Summary of all failing tests
2022-04-13T16:35:47.6438122Z [INFO] FAIL src/components/common/searchbar/Searchbar.test.js
2022-04-13T16:35:47.6446873Z [INFO]   ● Test Searchbar Component › Calls the callback function on keyup after timer runs out
2022-04-13T16:35:47.6447222Z [INFO] 
2022-04-13T16:35:47.6449817Z [INFO]     expect(jest.fn()).toBeCalled()
2022-04-13T16:35:47.6450056Z [INFO] 
2022-04-13T16:35:47.6452870Z [INFO]     Expected number of calls: >= 1
2022-04-13T16:35:47.6455791Z [INFO]     Received number of calls:    0
2022-04-13T16:35:47.6456017Z [INFO] 
2022-04-13T16:35:47.6459782Z [INFO]       59 |     await new Promise((r) => setTimeout(r, 1000));
2022-04-13T16:35:47.6460060Z [INFO]       60 |
2022-04-13T16:35:47.6463813Z [INFO]     > 61 |     expect(onUserSearch).toBeCalled();
2022-04-13T16:35:47.6466698Z [INFO]          |                          ^
2022-04-13T16:35:47.6470934Z [INFO]       62 |     expect(searchElement.value).toBe('Test Search');
2022-04-13T16:35:47.6473064Z [INFO]       63 |   });
2022-04-13T16:35:47.6475143Z [INFO]       64 | });
2022-04-13T16:35:47.6475357Z [INFO] 
2022-04-13T16:35:47.6480740Z [INFO]       at _callee$ (src/components/common/searchbar/Searchbar.test.js:61:26)
2022-04-13T16:35:47.6484063Z [INFO]       at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
2022-04-13T16:35:47.6488516Z [INFO]       at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
2022-04-13T16:35:47.6491393Z [INFO]       at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
2022-04-13T16:35:47.6494299Z [INFO]       at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
2022-04-13T16:35:47.6496870Z [INFO]       at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
2022-04-13T16:35:47.6499321Z [INFO] 
2022-04-13T16:35:47.6501639Z [INFO] 
2022-04-13T16:35:47.6505600Z [INFO] Test Suites: 1 failed, 140 passed, 141 total
2022-04-13T16:35:47.6508007Z [INFO] Tests:       1 failed, 477 passed, 478 total
2022-04-13T16:35:47.6509683Z [INFO] Snapshots:   0 total
2022-04-13T16:35:47.6510306Z [INFO] Time:        143.105 s
2022-04-13T16:35:47.6510905Z [INFO] Ran all test suites.
2022-04-13T16:35:47.9296329Z [INFO] error Command failed with exit code 1.
2022-04-13T16:35:47.9297114Z [INFO] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2022-04-13T16:35:47.9476337Z [INFO] error Command failed with exit code 1.```

File: openmetadata-ui/src/main/resources/ui/src/components/ManageTab/ManageTab.component.tsx [hunk 1]
```diff
@@ -415,6 +415,32 @@ const ManageTab: FunctionComponent<Props> = ({
...
+            </div>
+            <Button
+              className="tw-px-2 tw-py-0 tw-rounded tw-h-8 tw-self-center tw-shadow"
+              data-testid="delete-button"
+              size="custom"
+              theme="primary"
+              type="button"
+              variant="outlined"
+              onClick={handleOnEntityDelete}>
+              Delete this entity
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/ManageTab/ManageTab.component.tsx:438]:
```suggestion
              Delete this {entityType}
```

**Comment by vivekratnavel** [comment]
@Sachin-chaurasiya Merging this patch now, but can you take care of the code smells in your next patch?

**Pull request merged**